engine/src/components/reactor.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * reactor.h
+ *
+ * Copyright (c) 2001-2002 Daniel Horn
+ * Copyright (c) 2002-2019 pyramid3d and other Vega Strike Contributors
+ * Copyright (c) 2019-2023 Stephen G. Tuggy, Benjamen R. Meyer, Roy Falk and other Vega Strike Contributors
+ *
+ * https://github.com/vegastrike/Vega-Strike-Engine-Source
+ *
+ * This file is part of Vega Strike.
+ *
+ * Vega Strike is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Vega Strike is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Vega Strike. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// -*- mode: c++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+
+#ifndef REACTOR_H
+#define REACTOR_H
+
+#include "component.h"
+#include "energy_container.h"
+#include "energy_consumer.h"
+#include "resource/resource.h"
+
+class EnergyManager;
+
+class Reactor: public Component, public EnergyConsumer
```

Comment:
are all `EngeryConsumer` objects going to be `Component`s? I would think so. If so we might want to make `EnergyConsumer` a child of `Component` and then just have a single inheritance here.

Reply:
Not necessarily. A component is an upgrade-able and damageable module. One potential example of a consumer but not component is life support. 
I prefer to finish lib components and then do a cleanup pass for something like this. Constraining things now and finding out I need to revert is one of the reasons I've been working on this for six months.

Reply:
@royfalk that's fine - just thinking of the double inheritance and whether it's possible to avoid it. If not, okay, and we can always refactor it later too. Thanks for the explanation.